features/networking/egress-ip.feature

```diff
@@ -582,3 +586,195 @@ Feature: Egress IP related features
+    And the output should not contain "<%= cb.valid_ips[0] %>"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-40933
+  @admin
+  @destructive
+  Scenario: [sdn-1282] Manually EgressIPs assignments: if a pod is not on a node hosting an egressIP it is random which egressIP it will use
+    Given I save ipecho url to the clipboard
+    Given I store the masters in the :masters clipboard
+    Given I store the schedulable workers in the :workers clipboard
+    Given I store a random unused IP address from the reserved range to the clipboard
+
+    #Patch two egress ips to two nodes
+    Given I register clean-up steps:
+    """
+    as admin I successfully merge patch resource "hostsubnet/<%= cb.masters[0].name %>" with:
+      | {"egressIPs":null}   |
+    as admin I successfully merge patch resource "hostsubnet/<%= cb.workers[0].name %>" with:
+      | {"egressIPs":null}   |
+    """
+    Given as admin I successfully merge patch resource "hostsubnet/<%= cb.masters[0].name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ips[0] %>"] } |
+    Given as admin I successfully merge patch resource "hostsubnet/<%= cb.workers[0].name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ips[1] %>"] } |
+
+    # Create a pod and locate on specific node which is not an egress ip node
+    Given I have a project
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["template"]["spec"]["nodeName"] | <%= cb.workers[1].name %> |
+      | ["items"][0]["spec"]["replicas"]                     | 1                       |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+
+    # patch two egress ips to new project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ips[0] %>","<%= cb.valid_ips[1] %>"]} |
+
+    # Check the source IP randomly uses both egress ips
+    When I execute on the pod:
+      | bash | -c | for i in `seq 1 10`; do curl -s --connect-timeout 2 <%= cb.ipecho_url %> ; sleep 2;echo ""; done;  |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ips[1] %>"
+    And the output should contain "<%= cb.valid_ips[0] %>"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-40957
+  @admin
+  @destructive
+  Scenario: [SDN-1282] Auto EgressIPs assignments: if a pod is not on a node hosting an egressIP it is random which egressIP it will use
+    Given I save ipecho url to the clipboard
+    Given I store the masters in the :masters clipboard
+    Given I store the schedulable workers in the :workers clipboard
+    Given I store a random unused IP address from the reserved range to the clipboard
+    And evaluation of `IPAddr.new("<%= cb.subnet_range %>").to_s+"/"+IPAddr.new("<%= cb.subnet_range %>").prefix.to_s` is stored in the :valid_subnet clipboard
```

Comment:
Not necessary but we could use a lambda to do two operation on the same intermediate result.

```ruby
lambda { |i| "#{i.to_s}/#{i.prefix.to_s}" }.call(IPAddr.new("<%= cb.subnet_range %>"))

```

Reply:
Good idea, thanks Ross, updated it.